### DISCUSSION TITLE
No support for RasPi 4 2GB (Rev 1.5)?

### DISCUSSION THREAD

**Discussion by DerCoIsImHaus** [awawa-dev/HyperHDR#231] [General]
Hey there :)
I'm currently trying to get HyperHDR to work on my Raspi 4 2gb Rev 1.5 and I'm running into an issue...
when I select my LED-device (WS2812b) I get an Error message saying:

[LEDDEVICE_WS281X] Device disabled, device 'ws281x' signals error: 'Failed to open. Error message: Hardware revision is not supported'

is it possible that Rev 1.5 is not yet supported? It is as of now working with Hyperion NG nightly release, but I can't get it to work on HyperHDR :/

thanks for answers in advance :)

**Comment by awawa-dev** [comment]
Hi
It's caused by ws281x library and eventually as it is stated in FAQ official support for direct driving ws282b/sk6812 from Raspberry Pi was kicked out from HyperHDR ;) because there are much better and more stable solutions. But unofficially ;) I can tell you that you need newer ws281x library. Try to use newest installers (extract them from archive first) from the latest https://github.com/awawa-dev/HyperHDR/actions

**Comment by DerCoIsImHaus** [comment]
thanks for your answer :)

so does that mean I can't run it straight from the GPIO Pin to the strip? (sorry I'm fairly new to this ':D)
or do I just need to install this newer library for it to work?

**Comment by awawa-dev** [comment]
No, you can run it normally using GPIO Pin with a bit of luck (voltage level shifter helps a lot). You just need newest HyperHDR deb installer from github action artifacts. I keep ws281x  updated anyway, hope that version is sufficient. If not then I can refresh it again...but it doesn't comes free: that library needs some resources (quite precious ones: real time), if something goes wrong with the electrical installation then Rpi GPIO is first victim, there is also always a problem when new board or revision comes to the market (you are here) and of course root privileges but I see that you managed to pass this point ;)

**Comment by DerCoIsImHaus** [comment]
OK, great! I built it from source using your tutorial, so I think that would be the newest possible version :)
So for now I'll be waiting patiently :D 
Anyways thanks for taking your time, this is an awesome project you got there!

**Comment by awawa-dev** [comment]
Thanks!
You don't need to build them from sources yourself unless you prefer it that way: newest installers from github action always contain newest changes. Github automation takes care of that.
